### ISSUE TITLE
Full pre-activation implementation deviates from paper

### ISSUE THREAD

**Issue by JefferyRPrice** [raghakot/keras-resnet#26]
(This is a very nice implementation - thanks for sharing. I stumbled across here searching different ResNet implementations, especially those that have adopted the improved pre-activation stages.)

From the paper by He et al ([Identity Mappings in Deep Residual Networks](http://arxiv.org/abs/1603.05027)), there is a perhaps subtle implementation detail that doesn't appear to be implemented here. In particular, the paper states:

> When using the pre-activation Residual Units (Fig. 4(d)(e) and Fig. 5), we pay special attention to the first and the last Residual Units of the entire network. For the first Residual Unit (that follows a stand-alone convolutional layer, conv1), we adopt the first activation right after conv1 and before splitting into two paths; for the last Residual Unit (followed by average pooling and a fully- connected classifier), we adopt an extra activation right after its element-wise addition.

It seems that using the new pre-activation bottleneck stages as implemented here results in a `conv-bn-relu-bn-relu-...` entering into the bottleneck stages (back-to-back `bn-relu`) and that there is no activation, either `relu` or probably better `bn-relu`, heading into the average pooling.

FWIW, I have a [much less polished adaptation](https://gist.github.com/JefferyRPrice/c1ecc3d67068c8d9b3120475baba1d7e) that achieves 85.7% on CIFAR-10 with about 180K parameters (10 bottleneck stages, 31 total convolutional layers).

I might also suggest setting `bias = False` for any convolutional or dense layer followed by batchnorm since the batchnorm bias parameter makes that bias redundant.

**Comment by ahundt** [comment]
@JefferyRPrice could you make a pull request with these fixes?

**Comment by JefferyRPrice** [comment]
@ahundt I will. I think I have them done now in my own fork of this, but will submit PR once I verify things are working.

**Comment by JefferyRPrice** [comment]
@ahundt My fork mentioned above isn't quite correct and don't think I'm going to have time to make a PR.

My own resnet-like adaptation linked in my original issue used only bottleneck layers without any changes to the number of channels - I didn't account for that in my fork of this repo made some mistakes trying to fix the issues.

FWIW, I used my simpler resnet-like model to get 91.3% on CIFAR10. It's available as a [gist](https://gist.github.com/JefferyRPrice/c1ecc3d67068c8d9b3120475baba1d7e) and there's an image of the architecture (used `keras.utils.visualize.plot`).

**Comment by JefferyRPrice** [comment]
@ahundt Well I did have time and submitted PR #28

**Comment by ahundt** [comment]
@JefferyRPrice awesome, thanks!
